### PR TITLE
SASS-2216 Employment iterations

### PR DESCRIPTION
**Pull request by BenCosfordHMRC**
https://jira.tools.tax.service.gov.uk/browse/SASS-2216

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/employment/RemoveEmploymentControllerISpec.scala [hunk 1]
```diff
@@ -62,16 +62,18 @@ class RemoveEmploymentControllerISpec extends IntegrationTest with ViewHelpers w
   }
```
Comment (ina-tu) [inline it/controllers/employment/RemoveEmploymentControllerISpec.scala:63]: you could add a check for the inset text in one of the tests

File: test/utils/InYearUtilSpec.scala [hunk 2]
```diff
@@ -40,32 +40,32 @@ class InYearUtilSpec extends UnitTest {
...
+      inYearAction.inYear(yearCurrent, currentDate) shouldBe false
     }
   }
 
   "InYearAction.notInYear" when {
 
     def block: Future[Result] = { Future.successful(Results.Ok("Success Response")) }
 
-    "current date is before April 6th 2021 which means the tax year is 2020-21" should {
+    s"current date is before April 6th 2021 which means the tax year is 2020-21" should {
```
Comment (ina-tu) [inline test/utils/InYearUtilSpec.scala:68]: the tax years in the test names can be dynamic too

**Review by ina-tu — CHANGES_REQUESTED** [review]

File: app/views/employment/EmploymentSummaryView.scala.html [hunk 3]
```diff
@@ -78,14 +88,14 @@ <h2 class="govuk-heading-m govuk-!-margin-top-6">{headingMessage}</h2>
...
 
     @if(!isInYear) {
-        <p class="govuk-body"><a class="govuk-link" href="@EmploymentSummaryController.addNewEmployment(taxYear)">@messages("employment.employmentSummary.addAnotherEmployer")</a></p>
+        <p class="govuk-body"><a class="govuk-link" href="@EmploymentSummaryController.addNewEmployment(taxYear)">@messages(s"employment.employmentSummary.${if(employments.nonEmpty) "addAnotherEmployer" else "addAnEmployer"}")</a></p>
     }
 
-    <h2 class="govuk-label--m govuk-!-margin-top-9">@messages("employment.expenses.label")</h2>
+    <h2 id="expenses-h2" class="govuk-heading-m govuk-!-margin-top-9">@messages("employment.expenses.label")</h2>
     @if(expensesExist){
         <p class="govuk-body">@messages(s"employment.expenses.subheading")</p>
```
[the remark this answers is not in the document]
Comment (BenCosfordHMRC) [reply app/views/employment/EmploymentSummaryView.scala.html:100]: ah yes we had a chat about that and decided to only show it if there is data

[the remark this answers is not in the document]
Comment (ina-tu) [reply app/views/employment/EmploymentSummaryView.scala.html:100]: no problem

**Review by ina-tu — APPROVED** [review]
the other comments will be addressed in a follow-up PR

File: test/utils/InYearUtilSpec.scala (hunk 2 above)
Comment (BenCosfordHMRC) [reply test/utils/InYearUtilSpec.scala:68]: I left these one's as the tests below weren't dynamic ones

**Pull request merged**